src/runtime/pay-client.js

```diff
@@ -121,6 +126,41 @@ export class PayClient {
     pre.prefetch('https://clients2.google.com/gr/gr_full_2.0.6.js');
   }
 
+  /**
+   * Initializes Payments client.
+   */
+  initializePaymentsClient_() {
+    this.client_ = this.createClient_(
+      /** @type {!PaymentOptions} */
+      ({
+        environment: '$payEnvironment$',
+        'i': {
+          'redirectKey': this.redirectVerifierHelper_.restoreKey(),
+        },
+      }),
+      this.analytics_.getTransactionId(),
+      this.handleResponse_.bind(this)
+    );
+  }
+
+  /**
+   * Detects if the window is started from a Pay redirect by
+   * checking window's hash for Web Activities information.
+   */
+  pageIsInitializedFromPayRedirect_() {
+    const hash = this.win_.location.hash;
+    const hasWebActivitiesResponse = /^#__WA_RES__=/.test(hash);
```

Comment:
I am not a fan of referencing the internals of another library (namely web activities).  Philosophically,  this is brittle in case the library changes their implementation.  Presence of 'redirectEncryptedCallbackdata' should suffice.

Reply:
Removed the check of WA_RES. Relying on `redirectEncryptedCallbackdata` and `swgRequest` only.